cedar-lean/Cedar/Thm/Validation/Typechecker/HasAttr.lean

```diff
@@ -180,13 +189,18 @@ theorem type_of_hasAttr_is_sound {x₁ : Expr} {a : Attr} {c₁ c₂ : Capabilit
     simp [EvaluatesTo, h₆]
   case right =>
     rcases h₄ with ⟨ety, h₄⟩ | ⟨rty, h₄⟩ <;>
-    have ⟨_, v₁, h₆, h₇⟩ := ih h₁ h₂ h₄ <;>
+    simp [ResultType.typeOf, Except.map] at h₄ <;>
+    split at h₄ <;> simp at h₄ <;>
+    have ⟨ hl₄, _ ⟩ := h₄ <;>
+    rename_i h'₄ _ <;>
+    have ⟨_, v₁, h₆, h₇⟩ := ih h₁ h₂ h'₄  <;>
     simp [EvaluatesTo] at h₆ <;>
     simp [EvaluatesTo, evaluate] <;>
+    rw [hl₄] at h₇ <;>
     rcases h₆ with h₆ | h₆ | h₆ | h₆ <;> simp [h₆]
-    <;> try exact type_is_inhabited ty
-    · exact type_of_hasAttr_is_sound_for_entities h₁ h₂ h₃ h₄ h₆ h₇
-    · exact type_of_hasAttr_is_sound_for_records h₁ h₃ h₄ h₆ h₇
+    <;> try exact type_is_inhabited ty.typeOf
+    · exact type_of_hasAttr_is_sound_for_entities h₁ h₂ h₃ (by simp [h'₄, ResultType.typeOf, Except.map]; exact h₄) h₆ h₇
+    · exact type_of_hasAttr_is_sound_for_records h₁ h₃ (by simp [h'₄, ResultType.typeOf, Except.map]; exact h₄) h₆ h₇
```

Comment:
might be clearer to `apply` with an `_` in the place of the `(by )` expression, then put the `by` expression on the next line